tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -2731,6 +2731,37 @@ port::Status CudnnSupport::DoPrepareForConvolution(
   return port::Status::OK();
 }
 
+port::Status CudnnSupport::IntegerConvolutionSanityChecks(
+    const dnn::BatchDescriptor& input_descriptor,
+    const dnn::FilterDescriptor& filter_descriptor,
+    const dnn::BatchDescriptor& output_descriptor,
+    const absl::optional<dnn::AlgorithmDesc>& algo_desc) {
+  // Check algorithm.  Only _IMPLICIT_PRECOMP_GEMM is supported.
+  if (algo_desc.has_value() &&
```

Comment:
This check may fail with future cuDNN versions (which is also sane), right?

Reply:
That's correct, if cuDNN changes its behavior.  Putting all checks in one function makes maintenance a little easier.  
Actually, the transforms, such as pad_for_integer_convolution and setting layout to NHWC, will also have to update, when cuDNN changes.
If this is a big concern, we need at least one common predicate for the transforms and sanity check as they are based on the same assumption.  Once cuDNN changes in future, we can follow the predicate to make necessary updates.
How do you think?

Reply:
I added the sanity checks instead of replying on the status from the cuDNN APIs, because I found unsupported algorithm returns success.  I just received a confirmation from the cuDNN team this morning that it is a bug in the document.  I will revisit the sanity checks and whether it is really necessary.

Reply:
I have removed the sanity check as the CuDNN APIs work as design.  The document had a bug.